src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,134 @@
+        aux.setInverted(Ports.Shooter.AUX_INVERTED);
+        main.setSensorPhase(Ports.Shooter.IS_SENSOR_INVERTED);
+
+        main.setNeutralMode(NeutralMode.Coast);
+        aux.setNeutralMode(NeutralMode.Coast);
+
+        main.configForwardLimitSwitchSource(LimitSwitchSource.Deactivated, LimitSwitchNormal.Disabled);
+        main.configReverseLimitSwitchSource(LimitSwitchSource.Deactivated, LimitSwitchNormal.Disabled);
+
+        main.enableVoltageCompensation(true);
+        aux.enableVoltageCompensation(true);
+
+        main.configVoltageCompSaturation(Constants.VOLTAGE);
+        aux.configVoltageCompSaturation(Constants.VOLTAGE);
+
+        aux.follow(main);
+
+        Vector<N1> A = VecBuilder.fill(-Math.pow(G, 2) * Kt / (Kv * OMEGA * J)); //Change the amount of cells and rows
+        Vector<N1> B = VecBuilder.fill(G * Kt / (OMEGA * J));
+        LinearSystem<N1, N1, N1> stateSpace = new LinearSystem<>(A, B, Matrix.eye(Nat.N1()), new Matrix<>(Nat.N1(), Nat.N1()));
+        KalmanFilter<N1, N1, N1> kalman = new KalmanFilter<>(Nat.N1(), Nat.N1(), stateSpace,
+                VecBuilder.fill(MODEL_TOLERANCE),
+                VecBuilder.fill(ENCODER_TOLERANCE),
+                Constants.ROBOT_TIMEOUT
+        );
+        LinearQuadraticRegulator<N1, N1, N1> lqr = new LinearQuadraticRegulator<>(stateSpace, VecBuilder.fill(VELOCITY_TOLERANCE),
+                VecBuilder.fill(Constants.VOLTAGE),
+                Constants.ROBOT_TIMEOUT // time between loops, DON'T CHANGE
+        );
+        this.stateSpacePredictor = new LinearSystemLoop<>(stateSpace, lqr, kalman, Constants.VOLTAGE, Constants.ROBOT_TIMEOUT);
+    }
+
+    /**
+     * @return the velocity of the shooter. [rps]
+     * @see #setVelocity(double)
+     */
+    public double getVelocity() {
+        return rpsUnitModel.toVelocity(main.getSelectedSensorVelocity());
+    }
+
+    /**
+     * Set the velocity to apply by the motor.
+     *
+     * @param velocity the desired velocity at which the motor will rotate. [rps]
+     * @see #setPower(double)
+     */
+    public void setVelocity(double velocity) {
+        stateSpacePredictor.setNextR(VecBuilder.fill(velocity)); //r = reference
+        stateSpacePredictor.correct(VecBuilder.fill(getVelocity()));
+        stateSpacePredictor.predict(Constants.ROBOT_TIMEOUT); //every 20 ms
+
+        double voltageToApply = stateSpacePredictor.getU(0); // u = input, calculated by the input.
+        // returns the voltage to apply (between 0 and 12)
+        setPower(voltageToApply / Constants.VOLTAGE); // map to be between 0 and 1
+    }
+
+    /**
+     * Set the power to apply by the motor.
+     *
+     * @param power the power at which the motor will rotate. [percentage, between 0 and 1]
+     * @see #stop()
+     */
+    public void setPower(double power) {
+        main.set(TalonFXControlMode.PercentOutput, power, DemandType.ArbitraryFeedForward, Constants.Shooter.ARBITRARY_FEED_FORWARD);
```

Comment:
Because in testing we saw that some times there is a steady state error that the arbitrary feedforward handles with that pretty well

Reply:
Right, forgot about that.

Reply:
Have you asked?

Reply:
No, I don't even know what to ask.

Reply:
"Hey FRC Discord,
> in testing we saw that some times there is a steady state error (https://github.com/Galaxia5987/robot-2021/pull/8#discussion_r549341165)

do you know what could cause that?
Thanks!"

(And of course elaborate a bit, but if you managed to explain the situation to me, I believe you'll succeed in the Discord server too 😁)

Reply:
https://docs.ctre-phoenix.com/en/latest/ch16_ClosedLoop.html#do-i-need-to-use-arbitrary-feed-forward

> Units for the arbitrary feedforward term are [-1,+1].

Reply:
Yes, I know that. But how is it relevant?

Reply:
That's what you asked.

Reply:
Where did I ask where are the units for AFF?

Reply:
It's most likely because our moment of inertia is not the "real" value (I'm sorry about how I describe it, I can't remember how it's called), we estimate it from running the model

Reply:
I see, thanks. Maybe then you should use the gains from frc-characterization…
> I can't remember how it's called

Empirical?

Reply:
not technical, the word to describe it (btw the word is accurate), 

> I see, thanks. Maybe then you should use the gains from frc-characterization…

We check it later, with the other constants